doc/intergrations/intergration-with-datadog.md

```diff
@@ -0,0 +1,81 @@
+Datadog provides easy integration of log and metrics collection in the kubernetes environment.
+This document describes how to integrate the Datadog with Starrocks.
+
+> see https://docs.datadoghq.com/containers/kubernetes/installation/?tab=helm for more details.
+
+## 1. Install the Datadog chart
+
+If this is a fresh install, add the Helm Datadog repo:
+
+```bash
+helm repo add datadog https://helm.datadoghq.com
+helm repo update
+```
+
+Create a Kubernetes Secret to store your Datadog API key and app key:
+
+```bash
+export DD_API_KEY=xxx
+export DD_APP_KEY=yyy
+kubectl create secret generic datadog-secret --from-literal api-key=$DD_API_KEY --from-literal app-key=$DD_APP_KEY
+```
+
+Set the following parameters in your `datadog-values.yaml` to reference the secret:
+
+```bash
+datadog:
+  apiKeyExistingSecret: datadog-secret
+  appKeyExistingSecret: datadog-secret
+  site: <DATADOG_SITE>   # Replace <DATADOG_SITE> with your Datadog site.
+  logs:
+    enabled: true
+    containerCollectAll: true
+  prometheusScrape:
+    enabled: true
+    serviceEndpoints: true
+```
+
+Install the chart with the release name `datadog-agent`:
+
+```bash
+helm install datadog-agent  -f datadog-values.yaml datadog/datadog
+```
+
+## 2. Install or upgrade the starrocks cluster
+
+If this is a fresh install, add the Helm Starrocks Operator repo
+```bash
+helm repo add starrocks-community https://starrocks.github.io/starrocks-kubernetes-operator
+helm repo update
+```
+
+prepare the `sr-values.yaml` file for starrocks cluster:
+```bash
+# sr-values.yaml
+starrocksFESpec:
+  annotations:
+    # change the value of ad.datadoghq.com/fe.logs to your own value
+    ad.datadoghq.com/fe.logs: '[{"source": "fe","service": "starrocks","tags": ["env:test"]}]'
+    prometheus.io/path: "/metrics"
+    prometheus.io/port: "8030"
+    prometheus.io/scrape: "true"
```

Comment:
we shall be able to put this into FE service annotation instead of each POD annotation

And this is why I want helm chart/operator to do:
1. in order to collect log, `ad.datadoghq.com` annotation` needs to be added, and LOG_CONSOLE env var needs to be set
2. in order to collect metrics, `prometheus.io/*` annotation needs to be added, and the port shall be detected from the fe.conf or be.conf, fixed 8030/8040 may not work at all.